test/functional/finalization_vote.py

```diff
@@ -115,65 +115,63 @@ def run_test(self):
         generate_block(node0, count=2)
-        assert_equal(node0.getblockcount(), 35)
-        assert_finalizationstate(node0, {'currentDynasty': 4,
-                                         'currentEpoch': 7,
-                                         'lastJustifiedEpoch': 6,
-                                         'lastFinalizedEpoch': 5,
+        assert_equal(node0.getblockcount(), 25)
+        assert_finalizationstate(node0, {'currentDynasty': 3,
+                                         'currentEpoch': 5,
+                                         'lastJustifiedEpoch': 4,
+                                         'lastFinalizedEpoch': 4,
                                          'validators': 3})
         self.log.info('Finalizers voted on a configured block number')
 
         # test that finalizers can vote after configured epoch block number
         generate_block(node0, count=4)
-        assert_equal(node0.getblockcount(), 39)
+        assert_equal(node0.getblockcount(), 29)
         prev_tx = self.wait_for_vote_and_disconnect(finalizer=finalizer1, node=node0)
         self.wait_for_vote_and_disconnect(finalizer=finalizer2, node=node0)
         self.wait_for_vote_and_disconnect(finalizer=finalizer3, node=node0)
         generate_block(node0)
-        assert_equal(node0.getblockcount(), 40)
-        assert_finalizationstate(node0, {'currentDynasty': 5,
-                                         'currentEpoch': 8,
-                                         'lastJustifiedEpoch': 7,
-                                         'lastFinalizedEpoch': 6,
+        assert_equal(node0.getblockcount(), 30)
+        assert_finalizationstate(node0, {'currentDynasty': 4,
+                                         'currentEpoch': 6,
+                                         'lastJustifiedEpoch': 5,
+                                         'lastFinalizedEpoch': 5,
                                          'validators': 3})
         self.log.info('Finalizers voted after configured block number')
 
         generate_block(node0, count=4)
         prev_tx = finalizer1.decoderawtransaction(prev_tx)['txid']
 
         # test that node recognizes old and invalid votes.
-        tx = make_vote_tx(finalizer1, address1, node0.getblockhash(40), 5, 8, prev_tx)
-        assert_raises_rpc_error(-26, 'bad-vote-invalid', node0.sendrawtransaction, tx)
-        tx = make_vote_tx(finalizer1, address1, node0.getblockhash(40), 7, 9, prev_tx)
-        assert_raises_rpc_error(-26, 'bad-vote-invalid', node0.sendrawtransaction, tx)
-        tx = make_vote_tx(finalizer1, address1, node0.getblockhash(40), 5, 6, prev_tx)
-        assert_raises_rpc_error(-26, 'bad-vote-invalid', node0.sendrawtransaction, tx)
-        tx = make_vote_tx(finalizer1, address1, node0.getblockhash(40), 7, 6, prev_tx)
-        assert_raises_rpc_error(-26, 'bad-vote-invalid', node0.sendrawtransaction, tx)
+        # tx = make_vote_tx(finalizer1, address1, node0.getblockhash(30), 1, 2, prev_tx)
```

Comment:
Because this block creates votes which are surrounded/double votes for the previous ones and causes a floating condition in the test below. @frolosofsky is looking on it how to extract it to a different test

Reply:
I'll resolve this issue in this PR by switching the order of tests which are getting executed